osu.Game/IO/OsuStorage.cs

```diff
@@ -48,11 +48,17 @@ public void Migrate(string newLocation)
             var source = new DirectoryInfo(GetFullPath("."));
             var destination = new DirectoryInfo(newLocation);
 
+            if (source.FullName == destination.FullName)
+                throw new ArgumentException("Destination provided is already the current location", nameof(newLocation));
+
+            if (destination.FullName.Contains(source.FullName))
```

Comment:
This check is broken if there are two sibling directories
```
C:/somedir
C:/somedir2
```
as the test failure demonstrates (the paths returned by `FullName` don't have trailing slashes - at least on Windows, therefore there is a wrongly-matching common prefix).

Reply:
Fixed in #9027